File: backend/src/main/java/com/archive/entity/AuditLog.java
package com.archive.entity;

import com.baomidou.mybatisplus.annotation.TableField;
import com.baomidou.mybatisplus.annotation.TableName;
import com.baomidou.mybatisplus.extension.handlers.JacksonTypeHandler;
import lombok.Data;
import org.apache.ibatis.type.JdbcType;

import java.time.OffsetDateTime;
import java.util.Map;

/**
 * 操作审计日志。
 */
@Data
@TableName(value = "audit_logs", autoResultMap = true)
public class AuditLog {

    private Long id;

    private Long actorUserId;

    private String actorType;

    private String moduleName;

    private String operationType;

    private String businessType;

    private Long businessId;

    @TableField(typeHandler = JacksonTypeHandler.class, jdbcType = JdbcType.OTHER)
    private Map<String, Object> detail;

    private String ipAddress;

    private OffsetDateTime operatedAt;
}

File: frontend/src/types/audit-log.ts
import type { CursorParams } from './api'

/** 操作人类型 */
export type ActorType = 'internal' | 'public' | 'system'

/** 审计日志记录（DB §11.4 + §23.1） */
export interface AuditLog {
  id: number
  actorUserId: number
  actorType: ActorType
  /** 富字段：操作人姓名，便于展示 */
  actorName?: string
  moduleName: string
  /** 富字段：模块代号对应的中文标签（如 M14 -> 系统配置），便于展示 */
  moduleLabel?: string
  operationType: string
  /** 富字段：操作类型对应的中文标签（如 approve -> 审批通过），便于展示 */
  operationLabel?: string
  businessType?: string
  businessId?: number
  /** 富字段：当 businessType=archive 时关联档案的档号，便于展示 */
  archiveNo?: string
  /** JSONB 详情 */
  detail?: Record<string, unknown>
  ipAddress?: string
  operatedAt: string
}

/** 审计日志查询参数（§23.1） */
export interface AuditLogQuery extends CursorParams {
  actorUserId?: number
  actorType?: ActorType | ''
  moduleName?: string
  operationType?: string
  businessType?: string
  businessId?: number
  startedAt?: string
  endedAt?: string
}
